framework/src/headers/ip/v6/nf_macros.rs

```diff
@@ -0,0 +1,91 @@
+/*
+  Helper for inserting an SRH into a V6 Header and having it generate concrete
+  types for a range of segment lengths.
+
+  Macro parameters include:
+
+  o $segments - a vector of segments to insert as part of creating an SRH.
+  o $pkt - the current packet being operated on within a network function.
+  o $sr_info - SRH information, e.g. segments_left, tag, etc... that you want to
+               set upon insertion/creation.
+  o $prev_hdr_type - previous header type for SegmentRoutingHeader construction.
+  o [$len => $utype] - number-to-type pairs to allow for static writes of
+                       runtime numeric values, e.g. [1 => U1, 2 => U2]
+*/
+#[macro_export]
+macro_rules! srh_insert {
+    ($segments:expr, $pkt:expr, $sr_info:expr, $prev_hdr_type:ty, [$($len:expr => $utype:ty),*]) => {
```

Comment:
It's probably worth mentioning in the comments that the `pkt` really needs to be a `Packet<Ipv6Header>`. It's clearly the only time calling this would make sense, but is there anything ensuring that? 

Makes me wish we could do this with actual typed params instead of as a macro. What's the reason we have to use macros? Is it the var-length GenericArray situation?

Reply:
Ah, right, it can also be any Ipv6ExtHeader

Reply:
🌮